object.c

```diff
@@ -77,6 +77,10 @@ typedef enum {
     Map
 } type_t;
 
+#define NTYPES (Map + 1)
+#define TYPESIG(L, R) L*NTYPES+R
+#define CASE(L, R) case TYPESIG(L, R)
```

Comment:
Let's move these to just before addOperator() and friends so that they don't pollute the global namespace.  Also, don't forget to #undef them after the various *Operator() functions.

Reply:
Ok great idea!